sys/include/net/nanocoap.h

```diff
@@ -49,6 +49,7 @@ extern "C" {
  */
 #define NANOCOAP_NOPTS_MAX      (16)
 #define NANOCOAP_URI_MAX        (64)
+#define NANOCOAP_BLOCK_SZX_MAX  (6)
```

Comment:
Do we really need this nanocoap-specific definition? Why not redefine COAP_BLOCKWISE_SZX_MAX to 6. According to sec 2.2, the allowed values are 0 to 6.

Reply:
I've added some comments. Please let me know what you think and if we should handle this differently

Reply:
I get it now, but then what should this value represent? When I see 'SZX' in the name, I think of the value in the block option value (exponent - 4), not the value of the exponent. My comments in the implementation of coap_block2_init() show the confusion.

So, from my perspective the value of NANOCOAP_BLOCK_SZX_MAX should be 2 to map directly to the meaning of coap_block1_t.szx and COAP_BLOCKWISE_SZX_MAX. Using 2 for the value also would save the two subtractions in coap_block2_init().

Otherwise, it would make more sense to me to call this macro NANOCOAP_BLOCK_EXP_MAX, or something like that.

Reply:
I understand and agree with your arguments here. How about naming it `NANOCOAP_BLOCK_SIZE_EXP_MAX`? I'm against using the raw SZX value (2 in this case) because I want it to be as human readable as possible without too much room for mistakes.

Reply:
Name sounds fine. I would expect further development over negotiation of the size of the payload, so it will be helpful to think in those terms instead of the adjusted value encoded in the option.